src/MinterGateway.sol

```diff
@@ -881,12 +877,13 @@ contract MinterGateway is IMinterGateway, ContinuousIndexing, ERC712Extended {
      * @param newTimestamp_ The timestamp of the collateral update.
      */
     function _updateCollateral(address minter_, uint240 amount_, uint40 newTimestamp_) internal {
-        MinterState storage minterState_ = _minterStates[minter_];
-
-        uint40 lastUpdateTimestamp_ = minterState_.updateTimestamp;
+        uint40 minimumTimestamp_ = UIntMath.max40(
+            _minterStates[minter_].updateTimestamp,
+            uint40(block.timestamp) - updateCollateralInterval()
```

Comment:
I think it should be `newTimestamp_` and not `uint40(block.timestamp)` since `newTimestamp_` is used to set `minterState_.updateTimestamp`. So we should compare two validators timestamps and not a validator timestamp with a current `block.timestamp`.

https://github.com/MZero-Labs/protocol/blob/6c8d98526d77a75bd37fd466eec369211da78a36/src/MinterGateway.sol?plain=1#L896

Reply:
This is the earliest allowed timestamp that `newTimestamp_` can be, so it cannot possible be derived from `newTimestamp_`.